docs/guides/guide-for-writing-plugins.md

```diff
@@ -0,0 +1,71 @@
+---
+description: |
+  You need to use external code for performance 
+  reason or to avoid re inventing the wheel? This guide
+  explains how to integrate your code as a plugin.
+---
+
+# Guide for plugin development
+
+This page helps you get started building plugins for typst.
+
+Typst is capable of interfacing with plugins compiled to WebAssembly via it's `plugin` function.
+
+Once your program has been compiled to WebAssembly, it also need to respect [the protocol](#protocol). 
+
+Typst will run this code in isolation from your system which means printing and reading file will not be supported for security reasons.
+
+Many compilers will use the [wasi ABI](https://wasi.dev/) by default or as their only option (e.g. emscripten), which allows printing, reading file etc. This will not work with typst, you will need to either compile to a different target if possible or stub your library using a tool (see below).
```

Comment:
```suggestion
Many compilers will use the [wasi ABI](https://wasi.dev/) by default or as their only option (e.g. emscripten), which allows printing, reading files etc. This will not work with Typst, and you will need to either compile to a different target if possible or stub your library using a tool (see below).
```